.git-hooks/pre-push-scripts/validate_latest_release.bash

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+GIT_BRANCH="$(git rev-parse --abbrev-ref HEAD)"
+echo "➡️ On the $GIT_BRANCH branch."
+if [[ "$GIT_BRANCH" != main ]]; then
+	echo "➡️ Skip creating new release .. We only create release on puh to the main branch."
+	exit 0
+else
+	#
+	SECRETS_PATH="${PWD%/*/*}/.secrets"
+	SECRETS_FILE="$SECRETS_PATH/secrets.properties"	
+	GITHUB_TOKEN=$(grep -iR "^GITHUB_TOKEN" "$SECRETS_FILE" | awk -F "=" '{print $2}')
+	#
+	if [ -z "$GITHUB_TOKEN" ]
+	then
+		echo -e "\033[0;31m🚫 Invalid or empty GITHUB_TOKEN, pleasae make sure GITHUB_TOKEN exists in secrets.properties file in this path 👉 $SECRETS_FILE"
+	else
+		URL="https://api.github.com/repos/ahmadmssm/Test/releases"
+		#		
+		RESPONSE=$(curl -s -w %{http_code} -H Accept: application/vnd.github+json -H Authorization: Bearer $GITHUB_TOKEN $URL/latest)
+		HTTP_STATUS_CODE=$(tail -n1 <<< "$RESPONSE")  # get the last line.
+		HTTP_BODY=$(sed '$ d' <<< "$RESPONSE" )   # get all but the last line which contains the status code.
+		HTTP_BODY=$(echo $HTTP_BODY | sed -r 's/000000000//g')  # Remove leading zeros.
+		#echo "➡️ Status Code: $HTTP_STATUS_CODE"
+		#echo "➡️ Response: $HTTP_BODY"
+		#
+		if [[ "$HTTP_STATUS_CODE" -ne 200 ]] ; then 
+			echo -e "\033[0;31m🚫 Error, Could not fetch the latest GitHub release!"
+		else
+			LATEST_RELEASE=$(echo $HTTP_BODY | jq -r '.tag_name')
+			echo "➡️ Latest GitHub Release: $LATEST_RELEASE"
+			#
+			CURRENT_GIT_TAG=$(git describe --abbrev=0 --tags)
+			echo "➡️ Current GIT Tag: $CURRENT_GIT_TAG"
+			#
+			LOCAL_NPM_VERSION=$(cat "${PWD%/*/*}/package.json" | jq -r ".version")
+			echo "➡️ package.json version: $LOCAL_NPM_VERSION"
+			if [[ "$LATEST_RELEASE" != "$LOCAL_NPM_VERSION" ]] ; then
```

Comment:
I was rather thinking about controlling the version purely with `package.json` version.
With this logic it seems to me like:
- we read the latest GitHub tag version and check it against the version in `package.json` -> if they are different we show an error
- if the latest GitHub tag version is the same as the currently set git tag, we show an error
- else, we proceed with the the new version

I was rather thinking about this process:
- we check if the local package.json version is higher that the latest GitHub version -> if yes, then proceed with the new version (package.json one)
- if the package.json version is the same as the latest GitHub version -> do not create new tag nor publish
- if the package.json version is lower than the latest GitHub version -> show an error

We can discuss that approach on a call, because it may be that I am missing something.